### ISSUE TITLE
Dropdown headers are not read by screen readers

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3551]
## Describe the bug

The Dropdown headers are not read by screen reader users when they open the dropdown. 

This is because they are implemented as ARIA Menus, and as such they should not include headers. ARIA menus should only contain child elements with the following roles. 

- menuitem
- menuitemcheckbox
- menuitemradio

See https://www.w3.org/TR/wai-aria-practices-1.1/#menu.

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Activate the Dropdown Demos link under the Dropdown heading
- Activate the Dropdown 1 link under the 'Positioning Demo' heading
- Observe that the header is not announced
- Use the up and down arrow keys to navigate through the contents of the menu and notice that you cannot read the heading this way.

## Expected behavior

Implement correct aria and semantics.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

If we have to support Windows 10/JAWS 2019/Edge, I recommend to remove the header from the ARIA menu. If not, I recommend using aria-describedby to use the header as an accessible description for each related menuitem as seen in the "aria-describedby example" example under the 'Dropdown Examples' heading. While this will not cause the screen reader to read the heading when using the arrow keys, users will hear the heading as a description of each menu item as they read them using the arrow keys.
